### DISCUSSION TITLE
All permissions which apply on the user (inherited and direct)

### DISCUSSION THREAD

**Discussion by hesham-tech** [spatie/laravel-permission#2550] [Q&A]
When using
// All permissions which apply to the user (inherited and direct)
$user->getAllPermissions();
I get this error

Error: Call to a member function merge() on array in file C:\laragon\www\ecommerce\vendor\spatie\laravel-permission\src\Traits\HasPermissions.php on line 348

![image](https://github.com/spatie/laravel-permission/assets/94226077/ff0a7da4-3b4d-4bb3-a0ee-3c82f9f130ab)

**Comment by drbyte** [comment]
See #2551
